crates/autopilot/src/domain/fee/policy.rs

```diff
@@ -74,7 +74,7 @@ impl PriceImprovement {
             boundary::OrderClass::Limit => Some(domain::fee::Policy::PriceImprovement {
                 factor: self.factor,
                 max_volume_factor: self.max_volume_factor,
-                quote: quote.clone().into(),
+                quote: (*quote).into(),
```

Comment:
We can already remove the quote on the domain type (as this one isn't shared with solvers). We only need to keep the old format on the dto objects as solvers expect them to be there. That's one of the advantages in keeping domain and dto objects separate (we can refactor the domain regardless of the public API).

Not sure if it's feasible to clean up the PriceImprovement policy already in this PR.

Reply:
What is the best option to convert the domain to persistence when the order's quote is optional? Return another error if quote is empty? Convert PriceImprovement to Surplus?

https://github.com/cowprotocol/services/blob/8f587d7d2f17794063c7c17c22a8306abfe5f518/crates/autopilot/src/infra/persistence/dto/order.rs#L334-L362

Reply:
I see. This might be a lot of work (it cannot happen in practice, but if we remove the quote field soon, I guess it's ok to keep it on there for now)